pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -296,3 +323,35 @@ func (o createAuthInfoOptions) validate() error {
 
 	return nil
 }
+
+func (o *createAuthInfoOptions) checkPrompts() (err error) {
+	var result string
+
+	prompter := prompt.NewPrompter(bufio.NewReader(os.Stdin))
+
+	if o.promptUsername || o.promptCredentials {
+		result, err = prompter.Prompt("Username", prompt.ShowEcho, prompt.DontMask)
+		if err != nil {
+			return err
+		}
+		o.username.Set(result)
+	}
+
+	if o.promptPassword || o.promptCredentials {
+		result, err = prompter.Prompt("Password", prompt.DontShowEcho, prompt.Mask)
+		if err != nil {
+			return err
+		}
+		o.password.Set(result)
+	}
+
+	if o.promptToken {
```

Comment:
Does it make sense to provide both, for example, `--prompt-password` and `--prompt-token`? If not you probably need some validation for mutually-exclusive flags.